crates/runtime/tests/mysql/mod.rs

```diff
@@ -14,4 +14,127 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+use std::sync::Arc;
+
+use common::{get_mysql_conn, make_mysql_dataset, start_mysql_docker_container};
+use mysql_async::prelude::Queryable;
+
+use crate::init_tracing;
+
 pub mod common;
+
+use super::*;
+use app::AppBuilder;
+use mysql_async::{Params, Row};
+use runtime::Runtime;
+use tracing::instrument;
+
+const MYSQL_DOCKER_CONTAINER: &str = "runtime-integration-test-types-mysql";
+const MYSQL_PORT: u16 = 13316;
+
+#[instrument]
+async fn init_mysql_db(port: u16) -> Result<(), anyhow::Error> {
+    let pool = get_mysql_conn(port)?;
+    let mut conn = pool.get_conn().await?;
+
+    tracing::debug!("DROP TABLE IF EXISTS test");
+    let _: Vec<Row> = conn
+        .exec("DROP TABLE IF EXISTS test", Params::Empty)
+        .await?;
+
+    let _: Vec<Row> = conn
+        .exec(
+            "
+CREATE TABLE test (
+    id VARCHAR(36) PRIMARY KEY,
+    created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);",
+            Params::Empty,
+        )
+        .await?;
+
+    let _: Vec<Row> = conn
+        .exec(
+            "INSERT INTO test (id, created_at) VALUES ('5ea5a3ac-07a0-4d4d-b201-faff68d8356c', '2023-05-02 10:30:00-04:00');", Params::Empty
```

Comment:
Can we extend this to add more MySQL types and NULL?
https://github.com/spiceai/spiceai/blob/trunk/test/scripts/setup-data-mysql.sql